pyproject.toml

```diff
@@ -73,7 +73,6 @@ src = [
     "lib",
     "docs/src",
 ]
-target-version = "py39"
```

Comment:
I'm in favour of keeping a `target-version`. I do occasionally get burned by things that are not compatible with all our supported Python versions.

Any particular reason you would like to remove it?

If we do keep it then it would be another thing that should be included here:

https://github.com/SciTools/iris/blob/44bc4b856a1582c0f6a8bcf15d0ea6fa4df13f42/lib/iris/tests/test_coding_standards.py#L66-L71

Reply:
It's duplicate metadata that's not necessary due to the `requires-python` clause in the `pyproject.toml` file. `ruff` doesn't actually need it.

In fact, `repo-review` insists on removing it ... so it's something that we're going to remove now or later if we want to be compliant.

Reply:
My bad for not knowing that